sidecar/src/main.rs

```diff
@@ -136,4 +135,185 @@
     clippy::multiple_crate_versions, // caused by the dependency, can't be fixed
 )]
 
-fn main() {}
+use std::collections::HashMap;
+use std::fs::write;
+use std::path::{Path, PathBuf};
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use clap::{Parser, Subcommand};
+use tera::{Context, Tera};
+use tracing::debug;
+
+use xline_sidecar::config::{Backup, Config};
+use xline_sidecar::operator::Operator;
+
+/// Command line interface
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    /// The name of this node
+    #[arg(long)]
+    name: String, // used in xline and deployment operator to identify this node
+
+    /// Sub commands
+    #[command(subcommand)]
+    command: Commands,
+}
+
+/// Sub commands
+#[allow(variant_size_differences)] // required by clap
+#[derive(Subcommand, Debug)]
+enum Commands {
+    /// Run sidecar operator
+    Run {
+        /// Check health interval, default 20 [unit: seconds]
+        #[arg(long, default_value = "20")]
+        check_interval: u64,
+        /// Enable backup, choose a storage type, one of ['s3', 'pv']
+        #[arg(long)]
+        backup_type: Option<String>,
```

Comment:
We could use a enum value here for better help and error messages.

Reply:
done